cert.c

```diff
@@ -1,22 +1,30 @@
 #include "dnscrypt.h"
 
 struct SignedCert *
-cert_build_cert(const uint8_t *crypt_publickey, int cert_file_expire_days)
+cert_build_cert(const uint8_t *crypt_publickey, int cert_file_expire_days,
+                int use_xchacha20)
 {
     struct SignedCert *signed_cert = malloc(sizeof(struct SignedCert));
     if (!signed_cert)
         return NULL;
 
     memcpy(signed_cert->magic_cert, CERT_MAGIC_CERT, 4);
     signed_cert->version_major[0] = 0;
-    signed_cert->version_major[1] = 1;
+    if (use_xchacha20) {
+        signed_cert->version_major[1] = 2;
+    } else {
+        signed_cert->version_major[1] = 1;
+    }
     signed_cert->version_minor[0] = 0;
     signed_cert->version_minor[1] = 0;
 
     memcpy(signed_cert->server_publickey, crypt_publickey,
            crypto_box_PUBLICKEYBYTES);
     memcpy(signed_cert->magic_query, crypt_publickey,
            sizeof(signed_cert->magic_query));
+    if (use_xchacha20) {
```

Comment:
yeah, I totally missed that the intent could be to be able to generate a xchacha20 cert alongside a xsalsa20 one in which case it makes sense to bump it so to get a difference `magic_query`. Totally makes sense.